lib/usecase/sign_out_use_case.dart

```diff
@@ -0,0 +1,27 @@
+import 'dart:async';
+
+import 'package:injectable/injectable.dart';
+import 'package:survey_flutter_ic/api/exception/network_exceptions.dart';
+import 'package:survey_flutter_ic/api/persistence/auth_persistence.dart';
+import 'package:survey_flutter_ic/usecase/base/base_use_case.dart';
+
+import '../api/repository/auth_repository.dart';
+
+@Injectable()
+class SignOutUseCase extends NoParamsUseCase<void> {
+  final AuthRepository _repository;
+  final AuthPersistence _persistence;
+
+  const SignOutUseCase(this._repository, this._persistence);
+
+  @override
+  Future<Result<void>> call() async {
+    final token = await _persistence.accessToken ?? '';
+    return _repository
+        .signOut(token: token)
+        // ignore: unnecessary_cast
+        .then((token) => Success(null) as Result<void>)
+        .onError<NetworkExceptions>(
+            (exception, stackTrace) => Failed(UseCaseException(exception)));
+  }
```

Comment:
@hoangnguyen92dn I think we should clear the token in `persistence` when calling `signOut`

Reply:
@manh-t Indeed. Please allow me to add the clear tokens on the integration PR

https://github.com/hoangnguyen92dn/survey-flutter-ic/pull/76/files#diff-b5da62b96f31add08b94efede14c43dbe0c9c13eae063c3f54d51f864603b054R23